packages/commerce-sdk-react/src/hooks/useQuery.ts

```diff
@@ -1,18 +1,42 @@
 /*
- * Copyright (c) 2022, Salesforce, Inc.
+ * Copyright (c) 2023, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import {useQuery as useReactQuery, UseQueryOptions, QueryKey} from '@tanstack/react-query'
-import useAuthenticatedClient from './useAuthenticatedClient'
-import {IQueryFunction} from './types'
+import {useAuthorizationHeader} from './useAuthorizationHeader'
+import {ApiClient, ApiMethod, ApiOptions, MergedOptions} from './types'
+import {hasAllKeys, mergeOptions} from './utils'
 
-export const useQuery = <TData>(
-    queryKey: QueryKey,
-    fn: IQueryFunction<TData>,
-    queryOptions?: UseQueryOptions<TData, Error>
+export const useQuery = <
+    Options extends Omit<ApiOptions, 'body'>,
+    Client extends ApiClient,
```

Comment:
I think I kinda standardized on `<Client, Options>` being the usual ordering, I'll have to change this one.